Core/FileSystems/MetaFileSystem.cpp

```diff
@@ -270,28 +270,29 @@ std::string MetaFileSystem::NormalizePrefix(std::string prefix) const {
 	return prefix;
 }
 
-void MetaFileSystem::Mount(std::string prefix, IFileSystem *system)
-{
+void MetaFileSystem::Mount(std::string prefix, IFileSystem *system) {
 	std::lock_guard<std::recursive_mutex> guard(lock);
 	MountPoint x;
 	x.prefix = prefix;
 	x.system = system;
 	fileSystems.push_back(x);
 }
 
-void MetaFileSystem::Unmount(std::string prefix, IFileSystem *system)
-{
+void MetaFileSystem::Unmount(std::string prefix, IFileSystem *system) {
 	std::lock_guard<std::recursive_mutex> guard(lock);
 	MountPoint x;
 	x.prefix = prefix;
 	x.system = system;
 	fileSystems.erase(std::remove(fileSystems.begin(), fileSystems.end(), x), fileSystems.end());
 }
 
-void MetaFileSystem::Remount(IFileSystem *oldSystem, IFileSystem *newSystem) {
-	for (auto it = fileSystems.begin(); it != fileSystems.end(); ++it) {
-		if (it->system == oldSystem) {
-			it->system = newSystem;
+void MetaFileSystem::Remount(std::string prefix, IFileSystem *newSystem, bool delOldSystem) {
+	std::lock_guard<std::recursive_mutex> guard(lock);
+	for (auto &it : fileSystems) {
+		if (it.prefix == prefix) {
+			if (delOldSystem)
```

Comment:
Hm.  I think we're still leaking some of them, although I agree with making this delete them instead of `__UmdReplace()`.

Since this function takes ownership, I think it needs to handle deleting any systems that are no longer used.  I don't think it should expect the caller to know: for example, the old `disc0` also needs deleting.  We could do something like:

```c++
IFileSystem *removedSystem = nullptr;
for (auto &it : fileSystems) {
	if (it.prefix == prefix) {
		removedSystem = it.system;
		it.system = newSystem;
	}
}

bool delOldSystem = true;
for (auto &it : fileSystems) {
	if (it.system == removedSystem) {
		delOldSystem = false;
	}
}
```

Or something like that.  Probably even better would be to move to `std::shared_ptr` or otherwise for the system management...

-[Unknown]